### ISSUE TITLE
what should be the value of Dense()?

### ISSUE THREAD

**Issue by deshwalmahesh** [xuxingya/tf2crf#28]
You have used `Dense(9)`. I want to if 9 is the number of `tags` to find? How do you determine number og tags that I have to find? Also, Why `Dense` is not wrapped around `TimeDistributed`. Is it because the there is no need to wrap `Dense()` when working with `3-D` inputs?

How would the `CRF` layer know how many tags to find? And if 9 is the number of unique `y` tags, then can't you use a

**Comment by xuxingya** [comment]
The Dense layer before CRF Layer: Dense(units=num_of_tag) defines the number of tags.
And there is no need to wapper Dense with TimeDistributed, because the kernel will be broadcast to every other dims of the input. Refer to https://www.tensorflow.org/api_docs/python/tf/keras/layers/Dense.

**Comment by deshwalmahesh** [comment]
Oh yeah, I got that. Older versions of `Keras`  used to have that wrapped around. Now I don't see much of that. Thanks a lot for the quick answer. 

Also, I think you have used a modified version of `accuracy` and a `loss` function inside your `ModelWithCRFLoss` class so I don't have to use that. Right?

Also, number of tokens in each `X` are equal to no of tokens in corresponding `y`. Just conforming the things as this is the first time learning `CRF` for me.

**Comment by xuxingya** [comment]
You can directly use the predefined accuracy and loss, it's almost the same with the old keras contrib CRF. You can also override it if you want to usee your own metric.
Yes, CRF doesn't change the length of sequences.
